launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/ui/corebuild/CoreBuildMainTab.java

```diff
@@ -10,73 +10,51 @@
  *******************************************************************************/
 package org.eclipse.cdt.launch.ui.corebuild;
 
-import org.eclipse.cdt.launch.internal.ui.LaunchUIPlugin;
-import org.eclipse.core.resources.IProject;
-import org.eclipse.core.resources.IResource;
-import org.eclipse.core.runtime.CoreException;
+import org.eclipse.cdt.launch.internal.ui.LaunchMessages;
+import org.eclipse.cdt.launch.ui.CMainTab2;
 import org.eclipse.debug.core.ILaunchConfiguration;
-import org.eclipse.debug.core.ILaunchConfigurationWorkingCopy;
-import org.eclipse.debug.ui.AbstractLaunchConfigurationTab;
-import org.eclipse.swt.SWT;
-import org.eclipse.swt.layout.GridData;
-import org.eclipse.swt.layout.GridLayout;
 import org.eclipse.swt.widgets.Composite;
-import org.eclipse.swt.widgets.Label;
-import org.eclipse.swt.widgets.Text;
 
 /**
  * @since 9.1
  */
-public class CoreBuildMainTab extends AbstractLaunchConfigurationTab {
-
-	private Text projectName;
-
-	@Override
-	public void createControl(Composite parent) {
-		Composite comp = new Composite(parent, SWT.NONE);
-		comp.setLayout(new GridLayout());
-
-		Label label = new Label(comp, SWT.NONE);
-		label.setText("This launch configuration was automatically created.");
-		label.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
-
-		label = new Label(comp, SWT.NONE);
-		label.setText("Project:");
-
-		projectName = new Text(comp, SWT.READ_ONLY | SWT.BORDER);
-		projectName.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
-
-		setControl(comp);
-	}
+public class CoreBuildMainTab extends CMainTab2 {
 
+	/*
+	 * A Core Build launch configuration is created immediately upon the Core Build project creation.
+	 * It cannot be created by hand and it is not duplicatable and can't be renamed.
+	 * The launch configuration is tied to the project. The project name may not be changed.
+	 */
 	@Override
-	public void setDefaults(ILaunchConfigurationWorkingCopy configuration) {
-		// none
+	protected void createProjectGroup(Composite parent, int colSpan) {
```

Comment:
O: I'd prefer your previous patch where you used CoreBuildMainTab2 because this change will change the behaviour of the CoreBuildMainTab that I use.

Reply:
OK, no problem. Will change it back.

Reply:
Should I amend the commit and force push, or add commits to this PR?